File: herencia-dto/HerenciaDto-Java-IvanRC/src/main/java/models/Profesor.java
package models;

public final class Profesor extends Persona {
    String modulo;

    public Profesor(String nombre, String modulo) {
        super(nombre);
        this.modulo = modulo;
    }

    public String getModulo() {
        return modulo;
    }

    public void setModulo(String modulo) {
        this.modulo = modulo;
    }

    @Override
    public String toString() {
        return "Profesor{" +
                "modulo='" + modulo + '\'' +
                ", nombre='" + nombre + '\'' +
                '}';
    }
}

File: herencia-dto/SergioDeIscar-herencia-kotlin/SergioDeIscar-herencia-kotlin/src/main/kotlin/models/Profesor.kt
package models

import dto.PersonaDto
import java.io.Serializable

class Profesor(
    id: Long = 0,
    nombre: String,
    val modulo: String
): Persona(id, nombre), Serializable {
    override fun toCsvRow(): String {
        return "$id,$nombre,Profesor,null,$modulo\n"
    }

    override fun copy(id: Long, nombre: String): Persona {
        this.id = id
        return Profesor(id, nombre, modulo)
    }

    override fun toString(): String {
        return "Profesor(id=$id, nombre='$nombre', modulo='$modulo')"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Profesor) return false
        return id == other.id && nombre == other.nombre && modulo == other.modulo
    }
}
